var/spack/repos/builtin/packages/py-basemap/package.py

```diff
@@ -9,33 +9,37 @@ class PyBasemap(PythonPackage):
     """The matplotlib basemap toolkit is a library for plotting
     2D data on maps in Python."""
 
-    url = "https://github.com/matplotlib/basemap/archive/v1.2.0rel.tar.gz"
     homepage = "https://matplotlib.org/basemap/"
+    pypi = "basemap/basemap-1.4.1.zip"
 
     license("MIT")
 
-    version("1.2.1", sha256="3fb30424f18cd4ffd505e30fd9c810ae81b999bb92f950c76553e1abc081faa7")
+    version("1.4.1", sha256="730b1e2ff5eb31c73680bd8ebabc6b11adfc587cfa6832c528a8a82822e5a490")
+
+    variant("hires", default=False, description="Install hi-res data.")
 
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
 
+    depends_on("python@3.10:3.11", type=("build", "run"))
     # Per Github issue #3813, setuptools is required at runtime in order
     # to make mpl_toolkits a namespace package that can span multiple
     # directories (i.e., matplotlib and basemap)
     depends_on("py-setuptools", type=("build", "run"))
     depends_on("py-numpy@1.2.1:", type=("build", "run"))
-    depends_on("py-matplotlib@1.0.0:3.0.0,3.0.2:", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
```

Comment:
Per `requirements.txt`

```
matplotlib >= 1.5, < 2.0; python_version == "2.6"
matplotlib >= 1.5, < 3.0; python_version == "2.7"
matplotlib >= 1.5, < 2.0; python_version == "3.2"
matplotlib >= 1.5, < 2.0; python_version == "3.3"
matplotlib >= 1.5, < 3.0; python_version == "3.4"
matplotlib >= 1.5, < 3.9; python_version >= "3.5"
```

and the python version range, this should be constrained:

```suggestion
    depends_on("py-matplotlib@1.5:3.8 ", type=("build", "run"))
```

Reply:
You're right, I'll change that, thanks.